src/applications/vaos/appointment-list/pages/AppointmentsPage/index.unit.spec.js

```diff
@@ -567,14 +567,16 @@ describe('VAOS Page: AppointmentsPage', () => {
 
         await screen.findByRole('heading', { name: 'Appointments' });
 
-        expect(await screen.findByTestId('review-requests-and-referrals')).to
-          .exist;
+        waitFor(async () => {
```

Comment:
Curious why we need all of these new `waitFor` calls now in tests? Did using redux toolkit introduce some new race conditions or something?

Reply:
@JunTaoLuo I removed some of the waitFor that I added but some of them had to stay. The test were not passing. I think this is because adding the new hook changed the way the component gets rendered, so we have to do a waitFor.

Reply:
So it's jus the tests here an in [ReferralTaskCardWithReferral.unit.spec.js](https://github.com/department-of-veterans-affairs/vets-website/pull/34803/files#diff-bba24e3fd23707d9594f1232c522051c63bc951de069d19cf5dd9792450261e7)? I'm a little nervous that we don't completely understand the effects of using RTK on the rendering pipeline but I suppose if it's only a unit test problem it might be okay. I assume e2e tests work (Cypress) and also when you try this manually on a local build?

Reply:
@JunTaoLuo I think this is expected because I think before the component was looking at redux first and grabbing the data from there but in the new update it is using the API call instead. On the live site it should be utilizing cache if we already made the API call.